winforms/src/toga_winforms/libs/extensions.py

```diff
@@ -1,19 +1,31 @@
 import os
 import platform
+from pathlib import Path
 
 import clr
 
 #######################################################################
 # WebView2
 #######################################################################
-WEBVIEW2_DIR = os.path.join(os.path.dirname(__file__), "WebView2")
+WEBVIEW2_DIR = Path(__file__).parent / "WebView2"
 
-# The architecture-specific extension dlls folder must be in the path
-archpath = "x64" if platform.architecture()[0] == "64bit" else "x86"
-os.environ["Path"] = os.path.join(WEBVIEW2_DIR, archpath) + ";" + os.environ["Path"]
+# Derive the runtime's arch from the current machine's arch
+arch_path = {
+    "AMD64": "win-x64",
+    "x86": "win-x86",
+    "ARM64": "win-arm64",
+}[platform.machine()]
 
-clr.AddReference(os.path.join(WEBVIEW2_DIR, "Microsoft.Web.WebView2.Core.dll"))
-clr.AddReference(os.path.join(WEBVIEW2_DIR, "Microsoft.Web.WebView2.WinForms.dll"))
+# This specific filesystem layout is required for Windows Store Python.
+# While python.org Python will respect $PATH when searching for DLLs to load,
+# the Windows Store Python does not. Therefore, this filesystem layout is what
+# the WebView2 loader will search by default to find the runtime.
+# ref: https://learn.microsoft.com/en-us/microsoft-edge/webview2/concepts/distribution?tabs=dotnetcsharp#files-to-ship-with-the-app  # noqa: E501
+webview_runtime_dir = WEBVIEW2_DIR / f"runtimes/{arch_path}/native"
+os.environ["Path"] = f"{webview_runtime_dir}{os.pathsep}{os.environ['Path']}"
```

Comment:
Strictly speaking, adding this directory to `$PATH` is unnecessary since the WebView2 loader will eventually search the new `runtimes` directory. Although, I suppose that will only happen after it has searched all the directories in `$PATH`....so, leaving this in helps ensure (where possible) that Toga's shipped DLLs are what's actually loaded....and setting `$PATH` aligns with what pywebview is still doing.